Library/Homebrew/cask/cmd/zap.rb

```diff
@@ -32,14 +32,7 @@ def self.zap_casks(
         casks.each do |cask|
           odebug "Zapping Cask #{cask}"
 
-          if cask.installed?
-            if (installed_caskfile = cask.installed_caskfile) && installed_caskfile.exist?
-              # Use the same cask file that was used for installation, if possible.
-              cask = CaskLoader.load(installed_caskfile)
-            end
-          else
-            raise CaskNotInstalledError, cask unless force
-          end
+          raise CaskNotInstalledError, cask if !cask.installed? && !force
```

Comment:
Shouldn't we still try to load the installed caskfile and fall back to the one from the tap or the "dummy" cask we've created only if the installed caskfile is unreadable? This would ensure that the correct `zap` stanza is used and not one that's been updated for a newer version of the cask or one that's incomplete. Apologies if I've missed or misunderstood something.